RadixWallet/Clients/SubmitTransactionClient/SubmitTransactionClient+Live.swift

```diff
@@ -160,3 +160,77 @@ public enum TXFailureStatus: LocalizedError, Sendable, Hashable {
 		}
 	}
 }
+
+// MARK: TXFailureStatus.Reason
+extension TXFailureStatus {
+	public enum Reason: Sendable, Hashable, Equatable {
+		public enum ApplicationError: Equatable, Sendable, Hashable {
+			public enum WorktopError: Sendable, Hashable, Equatable {
+				case assertionFailed
+			}
+
+			case worktopError(WorktopError)
+		}
+
+		case applicationError(ApplicationError)
+		case unknown
+	}
+}
+
+/// Rudimentary parser combinator algo
+extension TXFailureStatus.Reason {
```

Comment:
I think this is trying to be a little bit "too smart", this breaks immediately when ever the error message String (`rawError`) changes format, making the solution brittle.  Which is why I proposed the simpler `.contains("AssertionFailed")` solution.